packages/dds/tree/src/shared-tree-core/sequenceIdUtils.ts

```diff
@@ -3,25 +3,43 @@
  * Licensed under the MIT License.
  */
 
+import { assert } from "@fluidframework/core-utils/internal";
 import { brand } from "../util/index.js";
 
 import type { SequenceId } from "./editManagerFormat.js";
 
+/**
+ * Compares two sequenceIds. Returns a negative number if a \< b, a positive number if a \> b, and 0 if a === b.
+ * Note that this handles cases where indexInBatch is Number.POSITIVE_INFINITY.
+ */
 export const sequenceIdComparator = (a: SequenceId, b: SequenceId): number =>
 	a.sequenceNumber !== b.sequenceNumber
 		? a.sequenceNumber - b.sequenceNumber
-		: (a.indexInBatch ?? 0) - (b.indexInBatch ?? 0);
+		: a.indexInBatch === b.indexInBatch
```

Comment:
Does this logic behave any differently than from before?

Reply:
It does for infinities: `Number.POSITIVE_INFINITY - Number.POSITIVE_INFINITY` returns `NaN`. The new logic will return 0.